### PR TITLE
NXP: Add I2C support for Kinetis KE1xF SoC and TWR-KE18F development board

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
This patch series adds support for I2C on the NXP Kinetis KE1xF SoC series and the TWR-KE18F development board through the following commits:
* soc: nxp: ke1xf: add I2C support
* boards: arm: twr_ke18f: add I2C support

### CODE REVIEW

File: boards/arm/twr_ke18f/doc/index.rst [hunk 1]
```diff
@@ -66,6 +66,12 @@ features:
 +-----------+------------+-------------------------------------+
 | FLASH     | on-chip    | soc flash                           |
 +-----------+------------+-------------------------------------+
+| I2C(M)    | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+| SENSOR    | off-chip   | fxos8700 polling;                   |
+|           |            | (fxos8700 trigger;)                 |
```
Comment (dbkinder) [inline boards/arm/twr_ke18f/doc/index.rst]: why the parentheses?  Maybe instead:
```
fxos8700 polling; trigger supported
with H/W mods explained below
```

File: boards/arm/twr_ke18f/doc/index.rst [hunk 2]
```diff
@@ -84,6 +90,28 @@ Serial Port
...
 
+Accelerometer and magnetometer
+==============================
+
+The TWR-KE18F board by default only supports polling the FXOS8700
+accelerometer and magnetometer for sensor values
+(``CONFIG_FXOS8700_TRIGGER_NONE=y``).
+
+In order to support FXOS8700 triggers (interrupts) the 0 ohm resistors
+``R47`` and and ``R57`` must be mounted on the TWR-KE18F board. The
```
Comment (dbkinder) [inline boards/arm/twr_ke18f/doc/index.rst]: Do you also have to change the value of CONFIG_FXOS8700_TRIGGER_NONE ?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/twr_ke18f/doc/index.rst (hunk 1 above)
Comment (henrikbrixandersen) [reply boards/arm/twr_ke18f/doc/index.rst]: @dbkinder Thank you for your feedback. I have updated the documentation with your suggested change.

File: boards/arm/twr_ke18f/doc/index.rst (hunk 2 above)
Comment (henrikbrixandersen) [reply boards/arm/twr_ke18f/doc/index.rst]: @dbkinder You can change it, if you want to use the triggers - but you do not have to.

Comment (dbkinder) [reply boards/arm/twr_ke18f/doc/index.rst]: This paragraph starts by saying what is needed to support triggers, so I'd think this paragraph should also mention that CONFIG_FXOS8700_TRIGGER_NONE would need to be changed to support triggers too, right?

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/twr_ke18f/doc/index.rst (hunk 2 above)
Comment (henrikbrixandersen) [reply boards/arm/twr_ke18f/doc/index.rst]: @dbkinder Good point. I have added that after the DTS snippet.

File: dts/arm/nxp/nxp_ke1xf.dtsi [hunk 3]
```diff
@@ -115,6 +117,30 @@
 			status = "disabled";
 		};
 
+		i2c0: i2c@40066000 {
+			compatible = "nxp,kinetis-lpi2c";
```
Comment (MaureenHelm) [inline dts/arm/nxp/nxp_ke1xf.dtsi]: @galak should we reuse the existing compatible `nxp,imx-lpi2c` instead?

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks

File: dts/arm/nxp/nxp_ke1xf.dtsi (hunk 3 above)
Comment (henrikbrixandersen) [reply dts/arm/nxp/nxp_ke1xf.dtsi]: I think that would be confusing at best. The recommended format for the DTS `compatible` string is `manufacturer,model`, which would not be correct if we used `nxp,imx-lpi2c`.

I suggest we either use `nxp,kinetis-lpi2c` as currently in this patch series - or that we go with the SoC series agnostic `nxp,lpi2c`.

We could also go with a combination of `compatible = "nxp,kinetis-lpi2c", "nxp,lpi2c"`.

Comment (MaureenHelm) [reply dts/arm/nxp/nxp_ke1xf.dtsi]: @galak has generally advised to reuse compatible strings if they have already been defined elsewhere, including Linux, for the same hardware ip. We already have precedent in the i.mx rt series using kinetis compatibles:
https://github.com/zephyrproject-rtos/zephyr/blob/ba4eae14ddea910777e11b1ac139fc147dc192c6/dts/arm/nxp/nxp_rt.dtsi#L243-L250

Unfortunately we don't yet support multiple compatibles in the yaml bindings.

Comment (galak) [reply dts/arm/nxp/nxp_ke1xf.dtsi]: Yeah, in the short term we should use the existing compatible.

Comment (galak) [reply dts/arm/nxp/nxp_ke1xf.dtsi]: Eventually we need to move to dt-schema which I think will solve the multiple compatible issue.

Comment (henrikbrixandersen) [reply dts/arm/nxp/nxp_ke1xf.dtsi]: @MaureenHelm @galak Thank you for your input on this. I have changed the commit to use the existing binding for now.

**Review by MaureenHelm — APPROVED** [review]
Thanks @henrikbrixandersen !

**Pull request merged**